src/net.cpp

```diff
@@ -2089,12 +2089,8 @@ void Relay(const CTransactionBase& tx, const CDataStream& ss)
     {
         if(!pnode->fRelayTxes)
             continue;
-        LOCK(pnode->cs_filter);
-        if (pnode->pfilter)
-        {
-            if (pnode->pfilter->IsRelevantAndUpdate(tx))
-                pnode->PushInventory(inv);
-        } else
+
+        if (pnode->nodeFilter.updateWith(tx))
```

Comment:
Seems a bit tricky that `updateWith(...)` returns `true` if `pfilter == nullptr`.
I would suggest to return `false` similar to the `insert(...)` method logic.
So, here we may do first: `if(!pnode->nodeFilter.isNull())) ...` and keep exactly the same flow as before.

Reply:
Done